server/modules/record/service/recordService.js

```diff
@@ -135,6 +140,42 @@ export const startCollectDataImportJob = ({ user, surveyId, filePath, deleteAllR
   return job
 }
 
+export const writeDataImportFromCSVTemplateToStream = async ({ surveyId, cycle, entityDefUuid, outputStream }) => {
+  const survey = await SurveyManager.fetchSurveyAndNodeDefsBySurveyId({ surveyId, cycle })
+  const entityDef = Survey.getNodeDefByUuid(entityDefUuid)(survey)
+  const exportModel = new CsvExportModel({
+    survey,
+    nodeDefContext: entityDef,
+    options: { includeFiles: false, includeCategoryItemsLabels: false, includeTaxonScientificName: false },
+  })
+  const dataRow = exportModel.columns.reduce((acc, column) => {
+    const { header, nodeDef, valueProp } = column
+    let value
+    if (nodeDef) {
+      const now = new Date()
+      const valuesByNodeDefType = {
+        [NodeDef.nodeDefType.boolean]: () => true,
+        [NodeDef.nodeDefType.code]: () => 'CATEGORY_CODE',
+        [NodeDef.nodeDefType.coordinate]: () =>
+          Points.toString(PointFactory.createInstance({ srs: 'EPSG:4326', x: 41.8830209, y: 12.4879562 })),
+        [NodeDef.nodeDefType.date]: () => DateUtils.formatDateISO(now),
+        [NodeDef.nodeDefType.decimal]: () => 123.45,
+        [NodeDef.nodeDefType.file]: () => '', // TODO support file attribute import
+        [NodeDef.nodeDefType.integer]: () => 123,
+        [NodeDef.nodeDefType.taxon]: () => 'TAXON_CODE',
+        [NodeDef.nodeDefType.text]: () => 'Text',
+        [NodeDef.nodeDefType.time]: () => DateUtils.formatTime(now.getHours(), now.getMinutes()),
```

Comment:
maybe this can go out of the service like template

Reply:
you are right... too many things there ;)